### ISSUE TITLE
Blank screen after new installation 

### ISSUE THREAD

**Issue by paulcarton** [PodcastGenerator/PodcastGenerator#745]
Podcast Generator Version: master and 3.2.9
Webserver: apache2
PHP Version: 8.1.2

No Config available

**Podcast Generator Config**

```
INSERT THE **CENSORED** CONFIG HERE
``` 

**Description of your problem:**
Hi everybody,

I'm sorry to seem to bringing up an old issue but I am being confronted with a blank screen after installation. 
After going to the ip-address of the installation, the welcome page comes up, then the screen saying that the media, images and other files are writable and continues to asking for an administrator name and password. When these are entered and I click continue I get a blank screen with the address "ip-address/setup/step3.php?create=1". It does not move on and when I refresh the site I get the address "ip-address/index.php" which is also blank.
I have spent a lot of time and effort trying to get this to work but I always end up with the same result. 
I have looked at and followed the Install from Scratch for both versions master and 3.2. I have set up my machine with both Ubuntu 22.04 and 23.04, checked both the apache and php installation on each and still end up at a blank screen. I checked Issue 272 and followed the instructions, checked that php-gettext and php-xml are installed and running (as can be seen in the info.php output: 
GetText 
Support - enabled 
and 
xml
XML Support - active
XML Namespace Support - active
libxml2 Version - 2.9.13)
The file ownership and permissions were set according to the instructions and the ports were opened in the firewall. 
As mentioned, it works to a point (first steps) but then turns a blank screen as soon as the initial setup is through.
What is also very confusing is that there is no config.php file anywhere, either before or after installation so it seems that the createconfig.php in the setup directory was not called on at all. 
I am using Proxmox VE and have a nextcloud, tt-rss server and my home page running on linux containers, all with no problems and accessable both from LAN and over the internet (behind a nginx PS) so not having PodcastGenerator running is very disappointing. Absolutely most annoying is that a previous (very old) version ran very well but I had to update the platform and I wanted to update the PodcastGenerater installation as well (instead of just restoring the old version) but this is not working.
Any help would be very much appreciated.
Thanks in advance and kind regards,
Paul Carton

**Comment by paulcarton** [comment]
Hallo,

just an update....

I saw that there was a composer.json file in the root of PG so I installed composer and made it globally accessible and ran the installation again. I had exactly the same problem as mentioned above but with the exception that a config.php file was generated. 
Here is the "censored" content of that config.php file:
Podcast Generator Config

INSERT THE **CENSORED** CONFIG HERE
<?php
$podcastgen_version = '3.2.9'; // Version

$first_installation = 1692262418;

CENSORED installationKey

$scriptlang = 'en_US';

CENSORED url

$absoluteurl = '/var/www/html/'; // The location on the server

$theme_path = 'themes/default/';

$upload_dir = 'media/'; // 'media/' the default folder (Trailing slash required). Set chmod 755

$img_dir = 'images/'; // (Trailing slash required). Set chmod 755

$feed_dir = ''; // Where to create feed.xml (empty value = root directory). Set chmod 755

$max_recent = 'all'; // How many file to show in the home page

$recent_episode_in_feed = 'All'; // How many file to show in the XML feed (1,2,5 etc.. or 'All')

$episodeperpage = 10;

$enablestreaming = 'yes'; // Enable mp3 streaming? ('yes' or 'no')

$freebox = 'yes'; // enable freely customizable box

$enablepgnewsinadmin = 'yes';

$strictfilenamepolicy = 'no'; // strictly rename files (just characters A to Z and numbers) 

$categoriesenabled = 'yes';

$cronAutoIndex = 1; //Auto Index New Episodes via Cron

$cronAutoRegenerateRSS = 1; //Auto regenerate RSS via Cron

$indexfile = 'index.php';    // Path of the index file

$podcastPassword = '';       // Password to protect the podcast generator webpages, this will NOT protect the audio or XML files. Leave blank to disable.

$customtagsenabled = 'no';   // Advanced functionality for custom RSS tag input

$timezone = '';              // Timezone used for displaying dates and times

#####################
# XML Feed stuff

$podcast_guid = ''; // Globally unique identifier for your podcast

CENSORED podcast_title

CENSORED podcast_subtitle

CENSORED podcast_description

$podcast_cover = 'itunes_image.jpg';

CENSORED author_name

CENSORED author_email

# The e-mail of the technical admin of the podcast
$webmaster = 'podcastadmin@example.com';

$itunes_category[0] = 'Arts'; // iTunes categories (mainCategory:subcategory)
$itunes_category[1] = '';
$itunes_category[2] = '';

$link = '?name='; // permalink URL of single episode (appears in the <link> and <guid> tags in the feed)

$feed_language = 'en';

$feed_sort = 'timestamp'; // sort method used to order episodes in the feed (by timestamp or by season/episode number)

$feed_locked = ''; // podcast:locked status ('yes', 'no', '' for off)

CENSORED copyright

$feed_encoding = 'utf-8';

CENSORED explicit_podcast

$users_json = '{\"admin\": \"\$2y\$10\$YvsSxMU5fyUGv.sqi8wEiOauiWPWAJDDh4jUacO80m.Qhl9cRblVi\"}';

#####################
# WebSub

$feed_encoding = 'utf-8';

CENSORED explicit_podcast

$users_json = '{\"admin\": \"\$2y\$10\$YvsSxMU5fyUGv.sqi8wEiOauiWPWAJDDh4jUacO80m.Qhl9cRblVi\"}';

#####################
# WebSub

$websub_server = '';

#####################
# Podcast Index

$pi_api_key = "";
$pi_api_secret = "";

$pi_podcast_id = 0; // is the podcast in Podcast Index? This is its show ID there.

#####################
# WebSub

$websub_server = "https://pubsubhubbub.appspot.com/";

// END OF CONFIG
$websub_server = '';

#####################
# Podcast Index

$pi_api_key = "";
$pi_api_secret = "";

$pi_podcast_id = 0; // is the podcast in Podcast Index? This is its show ID there.

#####################
# WebSub

$websub_server = "https://pubsubhubbub.appspot.com/";

// END OF CONFIG


I changed the ownership and permissions of config.php as described in issue 272 but to no avail, the blank screen keeps turning up. I hope that this will assist in finding out why the blank screen keeps turning up and finding a solution to this issue.

Thanks in advance for any help.

Kind regards,

Paul Carton

**Comment by paulcarton** [comment]
Sorry, that did not seem to copy correctly so here it is as an attached text file (config-php.txt)



<?php
$podcastgen_version = '3.2.9'; // Version

$first_installation = 1692262418;

$installationKey = 'tIgzcwuP'

$scriptlang = 'en_US';

$url = 'http://192.168.178.64/';

$absoluteurl = '/var/www/html/'; // The location on the server

$theme_path = 'themes/default/';

$upload_dir = 'media/'; // 'media/' the default folder (Trailing slash required). Set chmod 755

$img_dir = 'images/'; // (Trailing slash required). Set chmod 755

$feed_dir = ''; // Where to create feed.xml (empty value = root directory). Set chmod 755

$max_recent = 'all'; // How many file to show in the home page

$recent_episode_in_feed = 'All'; // How many file to show in the XML feed (1,2,5 etc.. or 'All')

$episodeperpage = 10;

$enablestreaming = 'yes'; // Enable mp3 streaming? ('yes' or 'no')

$freebox = 'yes'; // enable freely customizable box

$enablepgnewsinadmin = 'yes';

$strictfilenamepolicy = 'no'; // strictly rename files (just characters A to Z and numbers) 

$categoriesenabled = 'yes';

$cronAutoIndex = 1; //Auto Index New Episodes via Cron

$cronAutoRegenerateRSS = 1; //Auto regenerate RSS via Cron

$indexfile = 'index.php';    // Path of the index file

$podcastPassword = '';       // Password to protect the podcast generator webpages, this will NOT protect the audio or XML files. Leave blank to disable.

$customtagsenabled = 'no';   // Advanced functionality for custom RSS tag input

$timezone = '';              // Timezone used for displaying dates and times

#####################
# XML Feed stuff

$podcast_guid = ''; // Globally unique identifier for your podcast

$podcast_title = 'Podcast Title';

$podcast_subtitle = 'Subtitle';

$podcast_description = 'A little description of your podcast.';

$podcast_cover = 'itunes_image.jpg';

$author_name = 'Podcast Generator User';

$author_email = 'podcastgenerator@example.com';

# The e-mail of the technical admin of the podcast
$webmaster = 'podcastadmin@example.com';

$itunes_category[0] = 'Arts'; // iTunes categories (mainCategory:subcategory)
$itunes_category[1] = '';
$itunes_category[2] = '';

$link = '?name='; // permalink URL of single episode (appears in the <link> and <guid> tags in the feed)

$feed_language = 'en';

$feed_sort = 'timestamp'; // sort method used to order episodes in the feed (by timestamp or by season/episode number)

$feed_locked = ''; // podcast:locked status ('yes', 'no', '' for off)

$copyright = 'All rights reserved';   // Your copyright notice (e.g CC-BY)

$feed_encoding = 'utf-8';

$explicit_podcast = 'no'; //does your podcast contain explicit language? ('yes' or 'no')

$users_json = '{\"admin\": \"\$2y\$10\$YvsSxMU5fyUGv.sqi8wEiOauiWPWAJDDh4jUacO80m.Qhl9cRblVi\"}';

#####################
# WebSub

$feed_encoding = 'utf-8';

$explicit_podcast = 'no'; //does your podcast contain explicit language? ('yes' or 'no')

$users_json = '{\"admin\": \"\$2y\$10\$YvsSxMU5fyUGv.sqi8wEiOauiWPWAJDDh4jUacO80m.Qhl9cRblVi\"}';

#####################
# WebSub

$websub_server = '';

#####################
# Podcast Index

$pi_api_key = "";
$pi_api_secret = "";

$pi_podcast_id = 0; // is the podcast in Podcast Index? This is its show ID there.

#####################
# WebSub

$websub_server = "https://pubsubhubbub.appspot.com/";

// END OF CONFIG
$websub_server = '';

#####################
# Podcast Index

$pi_api_key = "";
$pi_api_secret = "";

$pi_podcast_id = 0; // is the podcast in Podcast Index? This is its show ID there.

#####################
# WebSub

$websub_server = "https://pubsubhubbub.appspot.com/";

// END OF CONFIG

[config-php.txt](https://github.com/PodcastGenerator/PodcastGenerator/files/12367757/config-php.txt)

**Comment by paulcarton** [comment]
Hi everybody,
I reinstalled PG, created a dedicated PG config file for sites-available in apache2, activated it and disabled the default config. I installed compose and made sure that all references to the web root path were correct in all relevant *.config and index.* files and checked that the root directory and sub-directories permissions were set to 776 (just in case!) and then everything worked. With just one exception but that is a case for a different issue. 
Regards to anyone who might have read this.
Paul Carton
